polartools/diffraction.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) 2020, UChicago Argonne, LLC.
+# See LICENSE file for details.
+
+import numpy as np
+from pandas import DataFrame
+from lmfit.models import (
+    GaussianModel,
+    LorentzianModel,
+    VoigtModel,
+    LinearModel,
+    PseudoVoigtModel,
+)
+import matplotlib.pyplot as plt
+from .load_data import load_table
+
+
+def fit_peak(xdata, ydata, model="Gaussian", output=False):
+    """
+    Fit Bragg peak with model of choice: Gaussian, Lorentzian, Voigt, PseudoVoigt.
+
+    Uses lmfit (https://lmfit.github.io/lmfit-py/).
+
+    WARNING: This imposes constrains to the fit that are described in
+    :func: `polartools.process_data._generate_initial_guess`
+
+    Parameters
+    ----------
+    xdata : iterable
+        List of x-axis values.
+    yydata : iterable
+        List of y-axis values.
+    model:
+        fit model: Gaussian, Lorentzian, Voigt, PseudoVoigt
+    output:
+        Output fit parameters and plot data+fit for each scan.
+
+    Returns
+    -------
+    fit : lmfit ModelResult class
+        Contains the fit results. See:
+        https://lmfit.github.io/lmfit-py/model.html#the-modelresult-class
+    """
+
+    models = {
```

Comment:
Instead using `()` here, which creates or calls *all* of these, move the parentheses to when `peak_mod` is used:

```python

    models = {
        "Gaussian": GaussianModel,
        "Lorentzian": LorentzianModel,
        "Voigt": VoigtModel,
        "PseudoVoigt": PseudoVoigtModel,
    }
    try:
        peak_mod = models[model]

    try:
        peak_mod = models[model]
    except KeyError:
        raise ValueError(
            f"model = {model} is invalid. Only these values are accepted: "
            f"{list(models.keys())}"
        )
    background = LinearModel
    mod = peak_mod() + background()
```

Same for `background`.  Or do I misunderstand how this code works?  As written, it is obfuscated.

Reply:
This does not work. No idea why. It only works when using the brackets as I had it.